src/erlcloud_ddb2.erl

```diff
@@ -248,13 +261,42 @@ default_config() -> erlcloud_aws:default_config().
 -type key() :: maybe_list(in_attr()).
 -type attr_defs() :: maybe_list({attr_name(), attr_type()}).
 -type key_schema() :: hash_key_name() | {hash_key_name(), range_key_name()}.
+-type hash_key_name() :: attr_name().
+-type range_key_name() :: attr_name().
+-type read_units() :: pos_integer().
+-type write_units() :: pos_integer().
+
+-type index_name() :: binary().
+-type projection() :: keys_only |
+                      {include, [attr_name()]} |
+                      all.
+
+-type global_secondary_index_def() :: {index_name(), key_schema(), projection(), read_units(), write_units()}.
 
 -type return_value() :: none | all_old | updated_old | all_new | updated_new.
 
+-type expression() :: binary().
+-type expression_attribute_names() :: [{binary(), attr_name()}].
+-type expression_attribute_values() :: [{binary(), in_attr_value()}].
```

Comment:
I'm afraid not, neither of them is expression.
Attribute names act as placeholders for reserved or long names according to the document.
Attribute values are also placeholders. Obviously, typed or complex values couldn't be embed into expressions that easy.